earthfile2llb/withdockerrun.go

```diff
@@ -62,17 +62,14 @@ type WithDockerOpt struct {
 	ComposeServices []string
 }
 
-type withDockerRun struct {
+type withDockerRunTar struct {
```

Comment:
This file is a bit hard to follow due to the two very similar, but distinct implementations in here. Mind splitting this into two distinct files?